### PR TITLE
Use `features.ALL` for a dict of all features

### PR DESCRIPTION
**Pull request by AhmetNSimsek**

### CODE REVIEW

**Comment by AhmetNSimsek** [comment]
This change allows users to easily access the list of available features. (Related example is also updated.)
```python
import siibra
siibra.features.ALL
```
Output:
```python
{'CorticalProfile': siibra.features.basetypes.cortical_profile.CorticalProfile,
 'Feature': siibra.features.basetypes.feature.Feature,
 'CellDensityProfile': siibra.features.cellular.cell_density_profile.CellDensityProfile,
 'Tabular': siibra.features.basetypes.tabular.Tabular,
 'LayerwiseCellDensity': siibra.features.cellular.layerwise_cell_density.LayerwiseCellDensity,
 'BigBrainIntensityProfile': siibra.features.cellular.bigbrain_intensity_profile.BigBrainIntensityProfile,
 'LayerwiseBigBrainIntensities': siibra.features.cellular.layerwise_bigbrain_intensities.LayerwiseBigBrainIntensities,
 'ReceptorDensityProfile': siibra.features.molecular.receptor_density_profile.ReceptorDensityProfile,
 'ReceptorDensityFingerprint': siibra.features.molecular.receptor_density_fingerprint.ReceptorDensityFingerprint,
 'GeneExpressions': siibra.features.molecular.gene_expression.GeneExpressions,
 'RegionalConnectivity': siibra.features.basetypes.regional_connectivity.RegionalConnectivity,
 'FunctionalConnectivity': siibra.features.connectivity.functional_connectivity.FunctionalConnectivity,
 'StreamlineCounts': siibra.features.connectivity.streamline_counts.StreamlineCounts,
 'StreamlineLengths': siibra.features.connectivity.streamline_lengths.StreamlineLengths,
 'EbrainsDataFeature': siibra.features.external.ebrains.EbrainsDataFeature,
 'VolumeOfInterest': siibra.features.basetypes.volume_of_interest.VolumeOfInterest}
 ```

File: examples/03_data_features/000_matchings.py [hunk 1]
```diff
@@ -30,7 +30,7 @@
 # Available feature types are listed in the module `siibra.features`.
 # Most of these represent specifically supported data modalities, and will
 # be covered one by one in the next examples.
-print(siibra.features.Feature.SUBCLASSES) # TODO: This is a temporary fix. 
+siibra.features.ALL
```
Comment (dickscheid) [inline examples/03_data_features/000_matchings.py:33]: this is a good idea, easy to understand

File: siibra/features/basetypes/feature.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ class Feature:
     """
 
     SUBCLASSES: Dict[Type['Feature'], List[Type['Feature']]] = defaultdict(list)
+    ALL =  Dict[str, Type['Feature']]
```
[the remark this answers is not in the document]
Comment (AhmetNSimsek) [reply siibra/features/basetypes/feature.py:38]: Yes, I think you are right. I'll double-check and remove if unnecessary

**Comment by dickscheid** [comment]
I just realize that the suggested `siibra.features.ALL` cannot be used in `siibra.features.get` in the current form of a dict. I will suggest a modification of this PR.

**Pull request merged**